metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -142,6 +162,55 @@ public Analysis importAnalysis(String accession) {
     protected abstract Analysis extractStudyFromAnalysis(AnalysisType analysisType, Analysis analysis);
 
     public ReferenceSequence importReferenceSequence(String accession) {
+        ReferenceSequence referenceSequence = null;
+        try {
+            String assemblyXml = sraXmlRetrieverByAccession.getXml(accession);
+            AssemblyType assembly = sraAssemblyXmlParser.parseXml(assemblyXml, accession);
+            referenceSequence = referenceSequenceConverter.convert(assembly);
+            Taxonomy taxonomy = taxonomyExtractorFromReferenceSequence.getTaxonomy(assembly.getTAXON());
+            referenceSequence.setTaxonomy(taxonomy);
+            referenceSequence = referenceSequenceRepository.findOrSave(referenceSequence);
+        } catch (Exception exception) {
+            IMPORT_LOGGER.log(Level.SEVERE, "Encountered Exception for ReferenceSequence accession " + accession);
+            IMPORT_LOGGER.log(Level.SEVERE, exception.getMessage());
+        }
+        return referenceSequence;
+    }
+
+    private Set<String> getReferenceSequenceAccessions(AnalysisType analysisType) {
+        Set<String> referenceSequenceAccessions = new HashSet<>();
+        AnalysisType.ANALYSISTYPE analysisType1 = analysisType.getANALYSISTYPE();
+        // ENA records can contain a reference sequence in either of the three analysis categories:
+        // REFERENCE_ALIGNMENT, SEQUENCE_VARIATION, and PROCESSED_READS. All accessions are collected and returned.
+        if (analysisType1.isSetREFERENCEALIGNMENT()) {
+            referenceSequenceAccessions.add(
+                    // TODO: should we use getRefname() or getAccession()?
+                    getAccessionFromReferenceSequenceType(analysisType1.getREFERENCEALIGNMENT()));
+        }
+        else if (analysisType1.isSetSEQUENCEVARIATION()) {
+            referenceSequenceAccessions.add(
+                    getAccessionFromReferenceSequenceType(analysisType1.getSEQUENCEVARIATION()));
+        }
+        else if (analysisType1.isSetPROCESSEDREADS()) {
+            referenceSequenceAccessions.add(
+                    getAccessionFromReferenceSequenceType(analysisType1.getPROCESSEDREADS()));
+        }
+        referenceSequenceAccessions.remove(null);
```

Comment:
Agreed; and I also noticed an implementation flaw in that is should have been not `if / else if / else if` block, but rather just a series of `if`s. I'm not sure whether an ENA analysis can include multiple types in practice, but the parser class implementation (e. g. method is named `isSetREFERENCEALIGNMENT`, not `isREFERENCEALIGNMENT`) suggests that theoretically it can happen.

Rewrote this block to be a series of `if`s and also to avoid removing `null` afterwards